src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -197,6 +199,26 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     };
   }
 
+  override getAccountFromRawAccount(
+    rawAccount: IAccountRaw,
+    idx: number,
+    globalIdx: number,
+    seed: Uint8Array,
+  ): IAccount {
+    if (rawAccount.type !== ACCOUNT_TYPES.hdWallet) {
+      throw new Error(`Unsupported account type: ${rawAccount.type}`);
+    }
+
+    const hdWallet = this.getHdWalletAccountFromMnemonicSeed(seed, idx);
```

Comment:
With the `getAccountFromRawAccount` method I think we no longer need the `getHdWalletAccountFromMnemonicSeed` as the required abstract method in the `BaseProtocolAdapter`.